codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -789,35 +789,38 @@ trim x y = let mid = (x - 2) `div` 2
-describePicture (Circle _ r) = printf "circle { radius = %4f }" r
-describePicture (SolidCircle _ r) = printf "solidCircle { radius = %4f }" r
-describePicture (ThickCircle _ lw r) = printf "thickCircle { linewidth = %4f , radius = %4f }" lw r
-describePicture (Polyline _ pts) = printf "polyline { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])
-describePicture (ThickPolyline _ pts w) = printf "thickPolyline { points = %s, width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
-describePicture (Curve _ pts) = printf "curve { points = %s }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts])
-describePicture (ThickCurve _ pts w) = printf "thickCurve { points = %s, width = %4f }" (intercalate ", " [printf "(%4f, %4f)" x y | (x,y) <- pts]) w
-describePicture (Sector _ b e r) = printf "sector { startAngle = %.2g° ( %.2g radians) , endAngle = %.2g° ( %.2g radians), radius = %4f}" (180 * b / pi) b (180 * e / pi) e r
-describePicture (Arc _ b e r) = printf "arc { startAngle = %.2g° ( %.2g radians) , endAngle = %.2g° ( %.2g radians), radius = %4f}" (180 * b / pi) b (180 * e / pi) e r
-describePicture (ThickArc _ b e r w) = printf "thickArc { startAngle = %.2g° ( %.2g radians), endAngle = %.2g° ( %.2g radians), radius = %4f , width = %4f}" (180 * b / pi) b (180 * e / pi) e r w
+describePicture (Rectangle _ w h) = "rectangle { width = " ++ showShortFloat w ++ " , height = " ++ showShortFloat h ++ "}"
+describePicture (SolidPolygon _ pts) = "solidPolygon { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ " }"
+describePicture (SolidClosedCurve _ pts) = "solidClosedCurve { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ " }"
+describePicture (Polygon _ pts) = "polygon { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ " }"
+describePicture (ThickPolygon _ pts w) = "thickPolygon { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ ", width = " ++ showShortFloat w ++ " }"
+describePicture (SolidRectangle _ w h) = "solidRectangle { width = " ++ showShortFloat w ++ " , height = " ++ showShortFloat h ++ "}"
+describePicture (ThickRectangle _ lw w h) = "thickRectangle { linewidth = " ++ showShortFloat lw ++ " , width = " ++ showShortFloat w ++ ", height = " ++ showShortFloat h ++ " }"
+describePicture (ClosedCurve _ pts) = "closedCurve { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ " }"
+describePicture(ThickClosedCurve _ pts w) = "thickClosedCurve { points = " ++ intercalate ", " [ "(" ++ showShortFloat x ++ ", " ++ showShortFloat y ++ ")" | (x, y) <- pts ] ++ ", width = " ++ showShortFloat w ++ " }"
+describePicture (Circle _ r) = "circle { radius = " ++ showShortFloat r ++ "}"
```

Comment:
Here and in the next few: everywhere else, we leave a space before the closing brace.  Let's do the same here.